### PR TITLE
[bugfix] fix mergeRollupTask metrics

### PR DESCRIPTION
**Pull request by zhtaoxiang**
1. Fix two typos: bucketTimeMs -> bufferTimeMs
2. Only keep/update lowestLevelMaxValidBucketEndTimeMs instead of maxValidBucketEndTimeMs of all merge levels

### CODE REVIEW

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java [hunk 1]
```diff
@@ -660,7 +662,7 @@ private void resetDelayMetrics(String tableNameWithType) {
    * Reset the delay metrics for the given table name and merge level.
    *
    * @param tableNameWithType table name with type
-   * @param mergeLevel merge level
+   * @param mergeLevel merge levelM
```
Comment (snleee) [inline pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java:665]: typo?

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java [hunk 2]
```diff
@@ -617,19 +619,19 @@ private void createOrUpdateDelayMetrics(String tableNameWithType, String mergeLe
...
     Map<String, Long> watermarkForTable =
         _mergeRollupWatermarks.computeIfAbsent(tableNameWithType, k -> new ConcurrentHashMap<>());
-    _tableMaxValidBucketEndTimeMs.put(tableNameWithType, maxValidBucketEndTimeMs);
     watermarkForTable.compute(mergeLevel, (k, v) -> {
       if (v == null) {
         LOGGER.info(
             "Creating the gauge metric for tracking the merge/roll-up task delay for table: {} and mergeLevel: {}."
                 + "(watermarkMs={}, bufferTimeMs={}, bucketTimeMs={}, taskDelayInNumTimeBuckets={})", tableNameWithType,
-            mergeLevel, watermarkMs, bucketTimeMs, bucketTimeMs,
+            mergeLevel, watermarkMs, bufferTimeMs, bucketTimeMs,
```
Comment (snleee) [inline pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java:627]: good catch 👍

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java [hunk 3]
```diff
@@ -459,7 +462,7 @@ private long getValidBucketEndTimeMsForSegment(SegmentZKMetadata segmentZKMetada
     // the rounded segment end time of [10/1 00:00, 10/1 23:59] is 10/2 00:00. The rounded segment end time of
     // [10/1 00:00, 10/2 00:00] is 10/3 00:00
     long validBucketEndTimeMs = (segmentZKMetadata.getEndTimeMs() / bucketMs + 1) * bucketMs;
-    validBucketEndTimeMs = Math.min(validBucketEndTimeMs, (currentTimeMs - bucketMs) / bucketMs * bucketMs);
+    validBucketEndTimeMs = Math.min(validBucketEndTimeMs, (currentTimeMs - bufferMs) / bucketMs * bucketMs);
```
Comment (snleee) [inline pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java:465]: good catch 👍

**Review by snleee — COMMENTED** [review]
@zhtaoxiang What's the concern with computing `maxValidBucketEndTimeMs` for each merge level? Since we are emitting the delay metric per (table, merge level) pair, I think that we need to compute `maxValidBucketEndTimeMs` for each merge level.

@jtao15 Can you double check on this?

**Review by snleee — APPROVED** [review]
@zhtaoxiang Oh I get your point. Yes, we need to keep the lowest level only as we use the watermark for the lower level from the 2nd level. Thanks for the clarification. This PR looks good to me!

@jtao15 Can you confirm the change before I merge this?

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java (hunk 3 above)
Comment (jtao15) [reply pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java:465]: +1

**Comment by jtao15** [comment]
@zhtaoxiang @snleee Correct, the `maxValidBucketEndTimeMs` is only used for the lowest level. 

- For the lowest level, the delay is calculated as (maxValidBucketEndTimeMs - waterMark) / bucketTime
- For other levels, we do (lowerLevelWatermark - waterMark) bucketTime

@zhtaoxiang Thanks for catching this, this looks good to me.

**Review by jackjlli — APPROVED** [review]
LGTM

**Pull request merged**